packages/legacy/hooks/src/use-clipboard.ts

```diff
@@ -24,20 +24,22 @@ export interface UseClipboardOptions {
  * @see Docs https://chakra-ui.com/docs/hooks/use-clipboard
  */
 export function useClipboard(
-  text: string,
+  initialValue: string,
   optionsOrTimeout: number | UseClipboardOptions = {},
 ) {
   const [hasCopied, setHasCopied] = useState(false)
 
+  const [value, setValue] = useState(initialValue)
```

Comment:
Can we use a `ref` here instead of state?

```js
  const valueRef = useRef(initialValue)
```

and then write to the ref in the setValue method

```js
const setValue = ( newValue: string ) => {
   valueRef.current = newValue
}
```

Reply:
Yes absolutely :)

Reply:
Actually after I tried to use `useRef` instead of `useState` all ways I found ended up with the value not beeing updated properly. So I'd stick with `useState`, even though we got the rerender. @segunadebayo